app/src/main/java/com/babylon/wallet/android/domain/usecases/signing/SignTransactionUseCase.kt

```diff
@@ -66,23 +69,21 @@ class SignTransactionUseCase @Inject constructor(
     }
 
     data class Request(
-        private val manifest: TransactionManifestData,
+        val manifestData: TransactionManifestData,
+        val manifest: TransactionManifest,
```

Comment:
maybe we should rename the `TransactionManifestData` to something else that indicates this is a data class from the network request. Quite confusing now the way it is.

Maybe to keep the sargon name `UnvalidatedManifestData`?
or `TransactionManifestDto`?

Reply:
Well, `TransactionManifestData` contains both the network request data and the actual data validated through Sargon, for example, the `TransactionManifest` here is resulting from the analysis, which involved getting the receipt and generating the execution summary - so it's no longer unvalidated data. I agree that it's kind of confusing and we could separate the raw input data within the request from the already processed data after run from Sargon. Maybe this is something to keep in mind when adding the support for the sub-intents as it will involve a larger refactoring anyway?

Reply:
I think I based my comments on the latest code changes which I haven't pushed yet at that moment, could you please re-review including the changes I've just pushed? Maybe you have different thoughts now since I updated the code around `TransactionManifestData`.

Reply:
ok now makes sense to me 👌🏽

Reply:
> I agree that it's kind of confusing and we could separate the raw input data within the request from the already processed data after run from Sargon. Maybe this is something to keep in mind when adding the support for the sub-intents as it will involve a larger refactoring anyway?

yes i think a separation might needed when adding support of sub-intents. If we take into consideration that a sub-intent based on enclosed or not it will run a preview and the other not